src/app/routes/assets/verification-method/verification-method.component.ts

```diff
@@ -55,6 +57,15 @@ export class VerificationMethodComponent implements OnInit {
     if (this.isFormDisabled) {
       return;
     }
+
+    const publicKeyExists = this.dataSource.map(key => key.publicKeyHex === this.publicKey.value);
```

Comment:
What do you think about creating a validator that will check if putted value do not exist in list?

Reply:
Okay. You mean in a different function right?

Reply:
Yeah. I think validator is fine here, but you might need to create async validator. I'm not sure, never done something like this 😛

Reply:
I do not understand what you mean by async validator

Reply:
This: https://angular.io/api/forms/AsyncValidator 
But you can fix it without that:
1. Create pure function to validate that -> returns true/false.
2. assign value returned from that function to some property.
3. display message if value is duplicated.
4. block sending form if input value is wrong.
5. Adjust styles.
6. Maybe something more to do.